src/XliffTasks/Model/XlfDocument.cs

```diff
@@ -70,7 +71,17 @@ public override void Save(TextWriter writer)
         public bool Update(TranslatableDocument sourceDocument, string sourceDocumentId)
         {
             bool changed = false;
-            Dictionary<string, TranslatableNode> nodesById = sourceDocument.Nodes.ToDictionary(n => n.Id);
+            var nodesById = new Dictionary<string, TranslatableNode>();
+            foreach (var node in sourceDocument.Nodes)
+            {
+                if (nodesById.ContainsKey(node.Id))
+                {
+                    throw new Exception($"A document has a duplicate node with the id '{node.Id}'.");
```

Comment:
sourceDocumentId would be useful in the error message here. 

Also, use BuildErrorException. This is a user error and soe we shouldn't need to dump a stack as though it was a bug.

Reply:
Done.